AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1795,7 +1795,26 @@ - (void)_pendingNodeDidDisplay:(ASDisplayNode *)node
 // Helper method to summarize whether or not the node run through the display process
 - (BOOL)__implementsDisplay
 {
-  return _flags.implementsDrawRect || _flags.implementsImageDisplay || _flags.shouldRasterizeDescendants || _flags.implementsInstanceDrawRect || _flags.implementsInstanceImageDisplay;
+  return _flags.implementsDrawRect || _flags.implementsImageDisplay || _flags.shouldRasterizeDescendants ||
+         _flags.implementsInstanceDrawRect || _flags.implementsInstanceImageDisplay;
+}
+
+- (BOOL)__canClearContentsOfLayer
+{
+  // The layer contents should not be cleared in case the node is wrapping a UIImageView.UIImageView is specifically
+  // optimized for performance and does not use the usual way to provide the contents of the CALayer via the
+  // CALayerDelegate method that backs the UIImageView.
+  return !_flags.synchronous || _viewClass != [UIImageView class];
+}
+
+- (BOOL)__canCallNeedsDisplayOfLayer
+{
+  // Prevent calling setNeedsDisplay on a layer that backs a UIImageView. Usually calling setNeedsDisplay on a CALayer
+  // triggers a recreation of the contents of layer unfortunately calling it on a CALayer that backs a UIImageView
+  // it goes trough the normal flow to assign the contents to a layer via the CALayerDelegate methods. Unfortunately
+  // UIImageView does not do recreate the layer contents the usual way, it actually does not implement some of the
+  // methods at all instead it throws away the contents of the layer and nothing will show up.
+  return _flags.synchronous && _viewClass != [UIImageView class];
```

Comment:
@maicki thanks for the comprehensive fix. I have tested it in https://github.com/molon/ASDKIssue853 .
@appleguy worry was right. Unless cache the result of  `isKindOfClass`, only `_viewClass` is not enough.